### ISSUE TITLE
Authenticate doesn't work

### ISSUE THREAD

**Issue by Yigityakut** [playgameservices/play-games-plugin-for-unity#2176]
Hello, I have been trying to integrate this plugin to my games more than 5 months. I tried everything and still couldn't manage this plugin to work. I published 1 games and 2 more waiting just for this plugin. 

I have encountered with too many problems. The latest one is : 
` DllNotFoundException: gpg at (wrapper managed-to-native) GooglePlayGames.Native.Cwrapper.Builder:GameServices_Builder_Construct ()`

People are saying that 'Force Resolve' is working for them or 'Reimporting the plugin'. Nothing works for me so far.  When I try to 'Force Resolve' I got 2 errors. ==> 
1 - Gradle failed to fetch dependencies.

```
Failed to run 'C:\Users\Yigit\Documents\GPGTEST\Temp\PlayServicesResolverGradle\gradlew.bat -b "C:\Users\Yigit\Documents\GPGTEST\Temp\PlayServicesResolverGradle\PlayServicesResolver.scripts.download_artifacts.gradle" --no-daemon "-PANDROID_HOME=C:/AndroidSDK" "-PTARGET_DIR=C:\Users\Yigit\Documents\GPGTEST\Assets\Plugins\Android" "-PMAVEN_REPOS=file:///C:/Users/Yigit/Documents/GPGTEST/Assets/GooglePlayGames/Editor/m2repository" "-PPACKAGES_TO_COPY=com.google.games:gpgs-plugin-support:0.9.50"'
stdout:

ERROR: JAVA_HOME is not set and no 'java' command could be found in your PATH.

Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.

stderr:

exit code: 1

UnityEngine.Debug:LogError(Object)
Google.Logger:Log(String, LogLevel)
GooglePlayServices.PlayServicesResolver:Log(String, LogLevel)
GooglePlayServices.<GradleResolution>c__AnonStorey14:<>m__1F(Result)
GooglePlayServices.<GradleResolution>c__AnonStorey15:<>m__28()
GooglePlayServices.PlayServicesResolver:PumpUpdateQueue()
UnityEditor.EditorApplication:Internal_CallUpdateFunctions()

```

2-  Resolution failed
```
Failed to fetch the following dependencies:
com.google.games:gpgs-plugin-support:0.9.50


UnityEngine.Debug:LogError(Object)
Google.Logger:Log(String, LogLevel)
GooglePlayServices.PlayServicesResolver:Log(String, LogLevel)
GooglePlayServices.ResolverVer1_1:LogMissingDependenciesError(List`1)
GooglePlayServices.<DoResolutionUnsafe>c__AnonStorey18:<>m__26(List`1)
GooglePlayServices.<GradleResolution>c__AnonStorey14:<>m__1F(Result)
GooglePlayServices.<GradleResolution>c__AnonStorey15:<>m__28()
GooglePlayServices.PlayServicesResolver:PumpUpdateQueue()
UnityEditor.EditorApplication:Internal_CallUpdateFunctions()
```

I do not think that I can resolve this issues (I feel deeply pessimistic about this after 5 months ). 
I was wondering if someone can share a sample project with just 2 3 buttons for login and show a leaderboard. Thank you in advance for your help.

**Comment by JimmyDeemo** [comment]
OK, lets deal with the first error;

`ERROR: JAVA_HOME is not set and no 'java' command could be found in your PATH.`

You need to follow the instructions here; https://github.com/playgameservices/play-games-plugin-for-unity#additional-instructions-on-building-for-android-on-windows. Once you set the environment variable in windows you will need to restart your machine in order for it to take affect.

**Comment by Yigityakut** [comment]
@JimmyDeemo Thank you so much Jimmy for your response. Thanks to you 'Force Resolve' worked and now there are files at Plugins/Android directory. Also, for the first time I saw the Google Play Games logo when I called  `Social.localUser.Authenticate(success =>{});` . Yet, Social.localUser's authenticated property remains false. I tried to dig into PlayGamesPlatform code but couldn't find anything. What should I do now?  I attached 2 screenshots from debug. 
![signinbefore](https://user-images.githubusercontent.com/22503715/38827854-e864c87a-41bc-11e8-803a-65d0270c8103.png)
![signincallback](https://user-images.githubusercontent.com/22503715/38827855-e882ed00-41bc-11e8-9b89-ce41f08fb80a.png)

**Comment by JimmyDeemo** [comment]
I think what would be best is for you to run the `Minimal` sample provided by the plugin. Get a fresh project and import the GPG plugin, and then the Minimal sample from the unity packages within the repo. Then start following the instructions from https://github.com/playgameservices/play-games-plugin-for-unity#configure-your-game. This should allow you to get to a simple example of authenticating.

If you get that to work then you can use it to compare to your project to see where there might be differences.

**Comment by Yigityakut** [comment]
@JimmyDeemo I am testing it at a test project already. I have 3 buttons.  1. SignIn 2.ReportScore 3.Showleaderboard. Here is my PlayGamesScript and how I call functions.  PlayGamesScript on MainCamera , ButtonScript on Canvas. Still, Social.localuser is like empty. I believe that is why I can not log in. I can open a public repository to share the test project if you want to see it.

PlayGamesScript

`using GooglePlayGames;
using GooglePlayGames.BasicApi;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using UnityEngine;
using UnityEngine.SocialPlatforms;


namespace Assets
{
  public  class PlayGamesScript : MonoBehaviour
    {

        void Start()
        {

            PlayGamesClientConfiguration config = new PlayGamesClientConfiguration.Builder().Build();
            PlayGamesPlatform.InitializeInstance(config);
            PlayGamesPlatform.DebugLogEnabled = true;
            PlayGamesPlatform.Activate();
        
        }

        public static void SignIn() {


            Social.localUser.Authenticate((bool success) => {
                if (success)
                {
                    Debug.Log("success");
                }
                else
                {
                    Debug.Log("failed to login");

                }
            });
        }

        public static void ShowLeaderboard ()
        {
            PlayGamesPlatform.Instance.ShowLeaderboardUI(GPGSIds.leaderboard_leaderboard);
        }
        public static void UpdateScoreToLeaderboard(int score, string leaderboard)
        {


            Social.ReportScore(score,leaderboard, (bool success) => {
                if (success)
                {
                    Debug.Log("success");
                }
                else
                {
                    Debug.Log("failed to login");

                }
            });
        }
        
    }
}
`

ButtonScript

`using Assets;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class scr : MonoBehaviour
{


    // Use this for initialization
    void Start()
    {

    }

    // Update is called once per frame
    void Update()
    {

    }

    public void btnClick(string btnName)
    {
        switch (btnName)
        {
            case "1":
                PlayGamesScript.SignIn();
                break;
            case "2":
                PlayGamesScript.UpdateScoreToLeaderboard(15,GPGSIds.leaderboard_leaderboard);
                break;
            case "3":

                break;
            case "4":
                PlayGamesScript.ShowLeaderboard();
                break;

            default:
                break;
        }



    }
}
`

**Comment by Yigityakut** [comment]
@JimmyDeemo success is false in callback. I knew it but I wanted to be sure to try it again after work. Now I am sure. 

And here is logcat logs:  (Log is looking bad if I paste it like code. I had to paste it as text)
I opened logcat with this command : ` adb logcat -s Unity ActivityManager PackageManager dalvikvm DEBUG
`

When I press the SignIn button, logs start after *****************. I wanted to share logs before I pressed. 


04 - 18 19: 47: 36.406 27316 27337 I Unity: [Play Games Plugin DLL] 04 / 18 / 18 19: 47: 36 + 03: 00 DEBUG: Invoking user callback on game thread
04 - 18 19: 47: 36.406 27316 27337 I Unity: UnityEngine.DebugLogHandler: Internal_Log(LogType, String, Object)
04 - 18 19: 47: 36.406 27316 27337 I Unity: UnityEngine.DebugLogHandler: LogFormat(LogType, Object, String, Object[])
04 - 18 19: 47: 36.406 27316 27337 I Unity: UnityEngine.Logger: Log(LogType, Object)
04 - 18 19: 47: 36.406 27316 27337 I Unity: UnityEngine.Debug: Log(Object)
04 - 18 19: 47: 36.406 27316 27337 I Unity: GooglePlayGames.OurUtils. < d > c__AnonStorey0: < > m__0()(at C: \Users\ Yigit\ Documents\ GPGTEST\ Assets\ GooglePlayGames\ OurUtils\ Logger.cs: 59)
04 - 18 19: 47: 36.406 27316 27337 I Unity: GooglePlayGames.OurUtils.PlayGamesHelperObject: Update()(at C: \Users\ Yigit\ Documents\ GPGTEST\ Assets\ GooglePlayGames\ OurUtils\ PlayGamesHelperObject.cs: 135)
04 - 18 19: 47: 36.406 27316 27337 I Unity:
    04 - 18 19: 47: 36.406 27316 27337 I Unity: (Filename: C Line: 0)
04 - 18 19: 47: 36.406 27316 27337 I Unity:
    04 - 18 19: 47: 40.287 1443 2097 I ActivityManager: Start proc 27548: com.miui.weather2 / u10a45
for service com.miui.weather2 / .service.ScheduleCheckService caller = android
04 - 18 19: 47: 40.483 1443 2552 I ActivityManager: Killing 26956: com.android.defcontainer / u0a10(adj 906): empty #17


`Before SignIn`
`****************************************************************************`
`After SignIn`

04-18 19:47:59.926  1443  1478 I ActivityManager: START u0 {act= com.google.android.gms.games.SIGN_IN flg = 0x4000000 pkg = com.google.android.play.games cmp = com.google.android.play.games / com.google.android.gms.games.ui.signin.SignInActivity(has extras)
}
from uid 10016
04 - 18 19: 47: 59.936 27316 27316 I Unity: onPause
04 - 18 19: 47: 59.989 27316 27337 D Unity: Sensor: Accelerometer(1);
0.002396 / 0.00 s;
ICM20690 Accelerometer / InvenSense
04 - 18 19: 47: 59.999 27316 27316 I Unity: windowFocusChanged: false
04 - 18 19: 48: 00.107 1443 1604 I ActivityManager: Displayed com.google.android.play.games / com.google.android.gms.games.ui.signin.SignInActivity: +115 ms
04 - 18 19: 48: 04.244 27316 27316 I Unity: windowFocusChanged: true
04 - 18 19: 48: 04.253 27316 27316 I Unity: onResume
04 - 18 19: 48: 04.262 27316 27337 D Unity: Sensor: Accelerometer(1);
0.002396 / 0.00 s;
ICM20690 Accelerometer / InvenSense
04 - 18 19: 48: 04.268 27316 27337 D Unity: Choreographer available: Enabling VSYNC timing
04 - 18 19: 48: 04.271 27316 27337 I Unity: [Play Games Plugin DLL] 04 / 18 / 18 19: 48: 04 + 03: 00 DEBUG: Application is pausing, which disconnects the RTMP client.Leaving room.
04 - 18 19: 48: 04.271 27316 27337 I Unity: UnityEngine.DebugLogHandler: Internal_Log(LogType, String, Object)
04 - 18 19: 48: 04.271 27316 27337 I Unity: UnityEngine.DebugLogHandler: LogFormat(LogType, Object, String, Object[])
04 - 18 19: 48: 04.271 27316 27337 I Unity: UnityEngine.Logger: Log(LogType, Object)
04 - 18 19: 48: 04.271 27316 27337 I Unity: UnityEngine.Debug: Log(Object)
04 - 18 19: 48: 04.271 27316 27337 I Unity: GooglePlayGames.OurUtils. < d > c__AnonStorey0: < > m__0()(at C: \Users\ Yigit\ Documents\ GPGTEST\ Assets\ GooglePlayGames\ OurUtils\ Logger.cs: 59)
04 - 18 19: 48: 04.271 27316 27337 I Unity: GooglePlayGames.OurUtils.PlayGamesHelperObject: Update()(at C: \Users\ Yigit\ Documents\ GPGTEST\ Assets\ GooglePlayGames\ OurUtils\ PlayGamesHelperObject.cs: 135)
04 - 18 19: 48: 04.271 27316 27337 I Unity:
    04 - 18 19: 48: 04.271 27316 27337 I Unity: (Filename: C Line: 0)
04 - 18 19: 48: 04.271 27316 27337 I Unity:
    04 - 18 19: 48: 08.487 27316 27337 I Unity: [Play Games Plugin DLL] 04 / 18 / 18 19: 48: 08 + 03: 00 DEBUG: Starting Auth Transition.Op: SIGN_IN status: ERROR_NOT_AUTHORIZED
04 - 18 19: 48: 08.487 27316 27337 I Unity: UnityEngine.DebugLogHandler: Internal_Log(LogType, String, Object)
04 - 18 19: 48: 08.487 27316 27337 I Unity: UnityEngine.DebugLogHandler: LogFormat(LogType, Object, String, Object[])
04 - 18 19: 48: 08.487 27316 27337 I Unity: UnityEngine.Logger: Log(LogType, Object)
04 - 18 19: 48: 08.487 27316 27337 I Unity: UnityEngine.Debug: Log(Object)
04 - 18 19: 48: 08.487 27316 27337 I Unity: GooglePlayGames.OurUtils. < d > c__AnonStorey0: < > m__0()(at C: \Users\ Yigit\ Documents\ GPGTEST\ Assets\ GooglePlayGames\ OurUtils\ Logger.cs: 59)
04 - 18 19: 48: 08.487 27316 27337 I Unity: GooglePlayGames.OurUtils.PlayGamesHelperObject: Update()(at C: \Users\ Yigit\ Documents\ GPGTEST\ Assets\ GooglePlayGames\ OurUtils\ PlayGamesHelperObject.cs: 135)
04 - 18 19: 48: 08.487 27316 27337 I Unity:
    04 - 18 19: 48: 08.487 27316 27337 I Unity: (Filename: C Line: 0)
04 - 18 19: 48: 08.487 27316 27337 I Unity:
    04 - 18 19: 48: 08.489 27316 27337 I Unity: [Play Games Plugin DLL] 04 / 18 / 18 19: 48: 08 + 03: 00 DEBUG: AuthState == Unauthenticated calling auth callbacks with failure
04 - 18 19: 48: 08.489 27316 27337 I Unity: UnityEngine.DebugLogHandler: Internal_Log(LogType, String, Object)
04 - 18 19: 48: 08.489 27316 27337 I Unity: UnityEngine.DebugLogHandler: LogFormat(LogType, Object, String, Object[])
04 - 18 19: 48: 08.489 27316 27337 I Unity: UnityEngine.Logger: Log(LogType, Object)
04 - 18 19: 48: 08.489 27316 27337 I Unity: UnityEngine.Debug: Log(Object)
04 - 18 19: 48: 08.489 27316 27337 I Unity: GooglePlayGames.OurUtils. < d > c__AnonStorey0: < > m__0()(at C: \Users\ Yigit\ Documents\ GPGTEST\ Assets\ GooglePlayGames\ OurUtils\ Logger.cs: 59)
04 - 18 19: 48: 08.489 27316 27337 I Unity: GooglePlayGames.OurUtils.PlayGamesHelperObject: Update()(at C: \Users\ Yigit\ Documents\ GPGTEST\ Assets\ GooglePlayGames\ OurUtils\ PlayGamesHelperObject.cs: 135)
04 - 18 19: 48: 08.489 27316 27337 I Unity:
    04 - 18 19: 48: 08.489 27316 27337 I Unity: (Filename: C Line: 0)
04 - 18 19: 48: 08.489 27316 27337 I Unity:
    04 - 18 19: 48: 11.462 27316 27337 I Unity: failed to login
04 - 18 19: 48: 11.462 27316 27337 I Unity: UnityEngine.DebugLogHandler: Internal_Log(LogType, String, Object)
04 - 18 19: 48: 11.462 27316 27337 I Unity: UnityEngine.DebugLogHandler: LogFormat(LogType, Object, String, Object[])
04 - 18 19: 48: 11.462 27316 27337 I Unity: UnityEngine.Logger: Log(LogType, Object)
04 - 18 19: 48: 11.462 27316 27337 I Unity: UnityEngine.Debug: Log(Object)
04 - 18 19: 48: 11.462 27316 27337 I Unity: Assets.PlayGamesScript: < SignIn > m__0(Boolean)(at C: \Users\ Yigit\ Documents\ GPGTEST\ Assets\ PlayGamesScript.cs: 36)
04 - 18 19: 48: 11.462 27316 27337 I Unity: GooglePlayGames. < Authenticate > c__AnonStorey1: < > m__0(Boolean, String)(at C: \Users\ Yigit\ Documents\ GPGTEST\ Assets\ GooglePlayGames\ ISocialPlatform\ PlayGamesPlatform.cs: 383)
04 - 18 19: 48: 11.462 27316 27337 I Unity: GooglePlayGames.Native. < InvokeCallbackOnGameThread > c__AnonStorey1 `2:<>m__0() (at C:\Users\Yigit\Documents\GPGTEST\Assets\GooglePlayGames\Platforms\Native\NativeClient.cs:168)
04-18 19:48:11.462 27316 27337 I Unity   : GooglePlayGames.OurUtils.PlayGamesHelperObject:Update() (at C:\Users\Yigit\Documents\GPGTEST\Assets\GooglePlayGames\OurUtils\PlayGamesHelperObject.cs:135)
04-18 19:48:11.462 27316 27337 I Unity   :
04-18 19:48:11.462 27316 27337 I Unity   : (Filename: C Line: 0)
04-18 19:48:11.462 27316 27337 I Unity   :
04-18 19:48:11.473 27316 27337 I Unity   :  [Play Games Plugin DLL] 04/18/18 19:48:11 +03:00 DEBUG: Invoking user callback on game thread
04-18 19:48:11.473 27316 27337 I Unity   : UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
04-18 19:48:11.473 27316 27337 I Unity   : UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
04-18 19:48:11.473 27316 27337 I Unity   : UnityEngine.Logger:Log(LogType, Object)
04-18 19:48:11.473 27316 27337 I Unity   : UnityEngine.Debug:Log(Object)
04-18 19:48:11.473 27316 27337 I Unity   : GooglePlayGames.OurUtils.<d>c__AnonStorey0:<>m__0() (at C:\Users\Yigit\Documents\GPGTEST\Assets\GooglePlayGames\OurUtils\Logger.cs:59)
04-18 19:48:11.473 27316 27337 I Unity   : GooglePlayGames.OurUtils.PlayGamesHelperObject:Update() (at C:\Users\Yigit\Documents\GPGTEST\Assets\GooglePlayGames\OurUtils\PlayGamesHelperObject.cs:135)
04-18 19:48:11.473 27316 27337 I Unity   :
04-18 19:48:11.473 27316 27337 I Unity   : (Filename: C Line: 0)
04-18 19:48:11.473 27316 27337 I Unity   :

**Comment by JimmyDeemo** [comment]
`04 - 18 19: 48: 08.487 27316 27337 I Unity: [Play Games Plugin DLL] 04 / 18 / 18 19: 48: 08 + 03: 00 DEBUG: Starting Auth Transition.Op: SIGN_IN status: ERROR_NOT_AUTHORIZED`

This line implies that the GPG plugin can't authorize your login. Please check that you have you have configuered things properly;.https://github.com/playgameservices/play-games-plugin-for-unity#configure-your-game. This would include, but is not limited to, making sure that the package name of the app matches the one used to create the valuse in the Google console and ensuring that you have ran the 'Setup' from within the play games configuration window.

The code itself looks sound, implying a configuration issue.

**Comment by Yigityakut** [comment]
@JimmyDeemo Well, I am afraid I am going to wait someone else to answer until I publish my games :( 

Thank you very much for your time and your help man. I appreciate it.  You solved one of the problems with environment variables stuff.  Thank you.

**Comment by dmitry-lagun** [comment]
Hey @Yigityakut , try insert your Web App Client Id in GPG setup's window in unity and press setup button.

**Comment by Yigityakut** [comment]
@JimmyDeemo @dmitry-lagun  I just found the problem guys. When you build the game through the unity with `Build And Run` it can not authenticate. When I released the game in Google Developer Console, I got successfully logged in and I was able to report score and see my leaderboard. Thank you so much for your helps!

**Comment by glitchersgames** [comment]
I had this issue as well and it was a configuration issue so tried what @Yigityakut said and to go through the Alpha/Beta release on the Google Play store and while uploading I got this error saying that the signed certificate did not match the upload certificate.

This is something I need to fix but the log showed the fingerprints not matching so I added the 'bad' SHA1 for now to the Google Cloud Platform Console / Credentials (https://console.cloud.google.com/apis/credentials) and it's working for debugging now. I will remove it when I am done with active development.

![screenshot_37](https://user-images.githubusercontent.com/20207200/40365279-15fb39ba-5dcc-11e8-9b8a-0fb04ffca05d.jpg)
